pkg/reconciler/utils/configmap.go

```diff
@@ -57,8 +57,12 @@ func (r *ConfigMapReconciler) ReconcileConfigMap(obj runtime.Object, cm *corev1.
 	if err != nil {
 		return nil, err
 	}
-
-	if !equality.Semantic.DeepEqual(cm.BinaryData, current.BinaryData) || !equality.Semantic.DeepEqual(cm.Data, current.Data) {
+	if eqFunc == nil {
```

Comment:
How about exporting the default eqFunc so that callers can use it if they want, and don't allow a nil eqFunc?  I think this is cleaner to use.

Reply:
Done. Adjusted tests accordingly.